File: api-gateway/src/main/java/com/tylerkrauss/api_gateway/model/Product.java
package com.tylerkrauss.api_gateway.model;

import lombok.*;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Getter
@Setter
public class Product {
   private String id;
   private String skuCode;
   private String name;
   private String description;
   private BigDecimal price;
}

File: microservices-frontend/src/app/model/product.ts
export interface Product {
  id?: string;
  skuCode: string;
  name: string;
  description: string;
  price: number;
}

//Quantity and skuCode go to inventory service. skuCode and everything besides quantity go to product service.
export interface ProductWithInventoryRequest {
  product: Product;
  quantity: number;
}
